tests/import_schemas.py

```diff
@@ -0,0 +1,213 @@
+    """
+    def __init__(self, sourceFile):
+        self.sourceFile = sourceFile
+        with open(sourceFile) as sf:
+            self.schemaSource = sf.read()
+            self.schema = avro.schema.parse(self.schemaSource)
+        self.name = self.schema.name
+
+    def getFields(self):
+        """
+        Returns the list of avro fields sorted in order of name.
+        """
+        return sorted(self.schema.fields, key=lambda f: f.name)
+
+    def isSearchRequest(self):
+        """
+        Returns True if the class we are converting is a subclass of
+        SearchRequest, and False otherwise.
+        """
+        return re.search('Search.+Request', self.name) is not None
+
+    def isSearchResponse(self):
+        """
+        Returns True if the class we are converting is a subclass of
+        SearchResponse, and False otherwise.
+        """
+        return re.search('Search.+Response', self.name) is not None
+
+    def getValueListName(self):
+        """
+        Returns the name of the list used to store values in a page for
+        a SearchRequest subclass.
+        """
+        assert self.isSearchResponse()
+        names = [field.name for field in self.getFields()]
+        # We assume that there are exactly two fields in every
+        # SearchResponse: nextPageToken and the value list.
+        names.remove('nextPageToken')
+        assert len(names) == 1
+        return names[0]
+
+    def getEmbeddedTypes(self):
+        """
+        Returns the set of embedded types in this class.
+        """
+        # TODO need to clarify how we operate on Unions here. The current
+        # code will break when we move to schema version 0.6 as we are
+        # no longer assured that the first element of the union is null.
+        # This would be a good opportunity to tidy this up.
+        ret = []
+        if isinstance(self.schema, avro.schema.RecordSchema):
+            for field in self.getFields():
+                if isinstance(field.type, avro.schema.ArraySchema):
+                    if isinstance(field.type.items, avro.schema.RecordSchema):
+                        ret.append((field.name, field.type.items.name))
+                elif isinstance(field.type, avro.schema.RecordSchema):
+                    ret.append((field.name, field.type.name))
+                elif isinstance(field.type, avro.schema.UnionSchema):
+                    t0 = field.type.schemas[0]
+                    t1 = field.type.schemas[1]
+                    if (isinstance(t0, avro.schema.PrimitiveSchema) and
+                            t0.type == "null"):
+                        if isinstance(t1, avro.schema.RecordSchema):
+                            ret.append((field.name, t1.name))
+                    else:
+                        msg = "Schema union assumptions violated: {}.{}"
+                        raise Exception(msg.format(self.name, field.name))
+        return ret
+
+
+class SchemaGenerator(object):
+    """
```

Comment:
I disagree. "Class that generates a schema in Python code from Avro definitions.". This is not what it does. A `SchemaGenerator` is something that generates a schema. We don't generate schemas here.